### PR TITLE
Restores cryoxadone to previous functionality, plus extra healing when used with upgraded freezers/cryotubes

### LINKED ISSUE #14897: #14761 Secretly nerfed Cryoxadone
Inside the massive PR (#14761) Neersighted reduced the amount Cryoxadone heals per tick for normal damage by 2/3, lowered the rate it heals clone loss to a QUARTER of what it was, and reduced the amount of oxyloss it heals at cryo temperatures to a FIFTH of what it was.

At no point was this change mentioned in the PR or in the comments, presumably it went unnoticed along with the 115 other changed files. And I can't see how "tgui wires; refactor wire datums; more to come" relates to major balance changes on medical reagents.

Cryoxadone is now literally worse than conventional healing chems, the only reason to use it is because its one of the only chems that heal cloneloss. Styptic powder/Silver Sulf heal more brute/burn over time AND heal when applied, Salbutamol heals more oxyloss. Even trek chems in sleepers heal more.

There is now literally no reason to use a cryotube except when healing cloned people. EVER.

### PR DESCRIPTION
**Pull request by Buggy123**
Fixes #14897

### CODE REVIEW

**Comment by CPTANT** [comment]
It seems to me the entire PR #14761 should be reverted, it has undocumented and unrelated changes everywhere.

**Comment by neersighted** [comment]
You do realize that these are ifs, not else ifs?

**Comment by duncathan** [comment]
Guys let's all take a long hard look at the code and understand that there is not a single instance of `else` in the code you are reverting

**Comment by ChangelingRain** [comment]
It's not actually a nerf, actually; it heals at roughly the same rate as it did before, and it's not an else if, it'd be applying all of those heals if you're in cyro.

Oh, and it actually starts healing at a higher temperature now.

**Comment by Buggy123** [comment]
Huh... apparently they do stack. Still, unupgraded cryo heals half as much clone loss and less oxygen loss than it used to. That's a significant nerf, and it was hidden and completely undocumented in a unrelated PR.

**Comment by Buggy123** [comment]
Also WTH didn't this compile I literally copypasta'd the old code.

**Comment by Buggy123** [comment]
Pre nerf, cryo heals:
4 clone per tick
10 oxy per tick
3 brute per tick
3 burn per tick
3 tox per tick

Post nerf, while in un upgraded cryo, it heals:
2 clone per tick
7 oxy per tick
3 brute per tick
3 burn per tick
3 tox per tick

IF upgraded(never happens), cryo heals:
7 clone per tick
9 oxy per tick
5 brute per tick
5 burn per tick
5 tox per tick

It wasn't as large of a nerf as it seemed, sure, but it's still significant. Halved cloneloss heal rate is pretty big.

**Comment by neersighted** [comment]
The comments are milsleading, You usually hit the second level, and even the third with regular cryo. Its just slower to get there without better freezers. Realisticly someone will probably be fully healed if not in crit before stage 3 kicks in.

**Comment by Buggy123** [comment]
It still greatly nerfs ghetto cryo and the secret ice-cryoxadone technique.

**Comment by neersighted** [comment]
You do realize that with the way new cryo works (suspends environment/breath, does not heal directly) this is actually a nerf?

**Comment by neersighted** [comment]
Re: ghetto cryo

I asked people about moving all heal to the chem and none objected. If someone wants to balance drinking cryox/ghetto cryo go ahead, but this maintains previous cryo balance. Please maintain cryo balance if you buff ghetto cryo.

**Comment by Buggy123** [comment]
You said you moved all the healing to the chem, did cryotubes actually do any healing themselves previously? I was under the impression they didn't.

**Comment by neersighted** [comment]
They did, they adjusted brute loss, fire loss, and tox loss when chilled below 225 previously. The 100 was because freezers after being buffed could get to 225 too easily and thus cryo healed faster. Moving a few points to 100 meant time was the ~same.

**Comment by KazeEspada** [comment]
Neer also made freezers less awful so making cyro colder is easier.

**Comment by Buggy123** [comment]
Alright I adjusted it, below the previous activation temperature (270) it behaves exactly the same as old cryoxadone, cryotube or not.

If you get cold enough via cryo upgrades, it heals even faster. I increased the <100 clone loss from 5 to 6 so it heals a nice round 10 altogether, and increased oxy heal rate for <100 to 5 for consistency sake and for a nice round number. Really ~10 oxy heal is a lot increasing it to 15 doesn't change anything.

**Comment by neersighted** [comment]
Please retitle your PR. in addition, the crazy oxy heal is not needed. Cryo stops the heart below 225k, and the number was so high only to outheal a bug in blood code, which I fixed.

**Comment by neersighted** [comment]
@Buggy123 Please don't make changes to code on Github, or without understanding it. `T0C` is a define for `273`, which is 0 degrees Celsius in Kelvin.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -108,21 +108,21 @@
...
+		M.adjustCloneLoss(-2)
 		M.adjustOxyLoss(-5)
 		M.adjustBruteLoss(-1)
 		M.adjustFireLoss(-1)
 		M.adjustToxLoss(-1)
-	if(M.stat != DEAD && M.bodytemperature < 225) // At lower temperatures (cryo) the full effect is boosted
-		M.adjustCloneLoss(-1)
-		M.adjustOxyLoss(-2)
+ 		M.status_flags &= ~DISFIGURED
+	if(M.stat != DEAD && M.bodytemperature < 270) // At lower temperatures (cryo) the full effect is boosted
```
Comment (neersighted) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: BAHAHAHAHAHAHA

**Comment by Buggy123** [comment]
270 Is the exact temperature it used beforehand, and I wanted to restore its exact previous functionality + the bonus healing if you upgrade so...

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 1 above)
Comment (neersighted) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: I DED

**Comment by neersighted** [comment]
Please join coderbus if you would like to know why this is funny.

**Comment by duncathan** [comment]
Your heart's in the right place man but everything about this pr is dead wrong

**Pull request closed without merging**